public/static/app/perfdiscovery/PerfDiscoveryService.js

```diff
@@ -0,0 +1,207 @@
+  function extractStorageEngine(build, task) {
+    var storageEngine,
+        attrs = {
+          build: build,
+          task: task,
+        }
+
+    if (build == undefined) return {}
+
+    _.any(attrs, function(v, k) {
+      var match = v.match(seRe)
+      if (match) {
+        storageEngine = match[1]
+        attrs[k] = attrs[k].slice(0, match.start)
+        return true
+      }
+    })
+
+    // Set default
+    if(!storageEngine) storageEngine = '(none)'
+
+    return _.extend({storageEngine: storageEngine}, attrs)
+  }
+
+  function slug(data) {
+    return [
+      data.build,
+      data.storageEngine,
+      data.task, data.test,
+      data.threads
+    ].join(' - ') + 'T'
+  }
+
+  // :param item: test result data item
+  // :param receiver: dict which will receive processed results
+  // :param ctx: dict of {build, task, storageEngine} see `extractTasks`
+  // returns: could be ignored
+  function processItem(item, receiver, ctx) {
+    var parts = item.name.split('-')
+    // At some point we renamed the tests to remove -wiredTiger and -MMAPv1 suffixs.
+    // Normalize old names to match the new ones
+    if (parts.length == 2 && _.any('wWmM', _.bind(''.startsWith, parts[0]))) {
+      item.name = parts[0]
+    }
+
+    return _.chain(item.results)
+      .omit(_.isString)
+      .each(function(speed, threads) {
+        var extracted = extractStorageEngine(ctx.buildName, ctx.taskName)
+        var data = {
+          build: extracted.build,
+          task: extracted.task,
+          test: item.name,
+          threads: +threads,
+          speed: speed.ops_per_sec,
+          storageEngine: extracted.storageEngine,
+        }
+        receiver[slug(data)] =  data
+      })
+  }
+
+  // Processes `queryData` return value
+  // Populates dicts of current (now, baseline) and history result items
+  // returns dict of {now, baseline, history} test result items
+  function processData(promise) {
+    var now = {},
+        baseline = {},
+        history = {}
+
+    return promise.then(function(data) {
+      _.each(data.tasks, function(d) {
+        _.each(d.current.data.results, function(result) {
+          processItem(result, now, d.ctx)
+        })
+      })
+
+      _.each(data.tasks, function(d) {
+        _.each(d.baseline.data.results, function(result) {
+          processItem(result, baseline, d.ctx)
+        })
+      })
+
+      _.each(data.tasks, function(d) {
+        _.each(d.history, function(histItems) {
+          _.each(histItems.data.results, function(result) {
+            processItem(result, history, d.ctx)
+          })
+        })
+      })
+
+      return {
+        now: now,
+        baseline: baseline,
+        history: history,
+      }
+    })
+  }
+
+  // Calculates ratio for given test result `item` and `baseSpeed`
+  function ratio(item, baseSpeed) {
+    var ratio = item.speed / baseSpeed
+    if (item.speed >= 0) {
+      return ratio
+    } else {
```

Comment:
What's the reason for inverting the ratio? Wouldn't this cause inaccurate results?

Reply:
I added a comment. Because speed could be negative, and inverting the ratio is kind of normalization, I think.

Reply:
If that's the case this should be returning -1*ratio, not the inverse of it

Reply:
Here is the original code:
```python
    def ratio(self, speed=None):
        if speed is None:
            speed = self.speed

        if speed >= 0:
            return speed / self.base_speed
        else:
            # Negatives mean latency in ms. Invert the ratio so higher is better.
            return 1.0 / (speed / self.base_speed)
```
If perf team is happy with the behavior, I don't think I should change it.

Reply:
Do you have any sample data for this? Inverting a negative ratio will still be negative so I'm not sure this is working correctly or the comments might be misleading

Reply:
@john-m-liu 
https://docs.google.com/spreadsheets/d/1BClkNOnJNbktl_AeNX59ga37YcuoQT8pACLGjRQxlCE/edit#gid=608598973